### PR TITLE
Upper bound on SciKit-Learn to avoid freeze in causal

### LINKED ISSUE #1430: Causal Analysis getting stuck
This appears to be related to the latest release of SciKit-Learn 1.1.0. I've found that causal analysis can get stuck under certain circumstances. Using a debugger, this appears to happen inside SciKit-Learn's tree estimators, but I've not managed to dissect it much more than that. It was found in the RAI component testing.

For a repo case, see PR #1429

### PR DESCRIPTION
**Pull request by riedgar-ms**
## Description

Closes #1430 . Causal Analysis is not getting along well with SciKit-Learn v1.1.0. This change incorporates a test which works under 1.0.2 but gets stuck when running with 1.1.0.

## Checklist

- [x] I have added screenshots above for all UI changes.
- [x] Documentation was updated if it was needed.
- [x] New tests were added or changes were manually verified.

### CODE REVIEW

File: responsibleai/tests/causal/test_causal_general.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    full_data[target_feature] = y
+
+    data_train, data_test = train_test_split(
+        full_data, test_size=1000, random_state=96132, stratify=full_data[target_feature]
+    )
+
+    categorical_columns = ["Race", "Sex", "Workclass",
+                           "Marital Status", "Country", "Occupation"]
+
+    return target_feature, categorical_columns, data_train, data_test
```
Comment (gaugup) [inline responsibleai/tests/causal/test_causal_general.py:25]: can you add the shap dataset into common_utils.py?

https://github.com/microsoft/responsible-ai-toolbox/blob/main/responsibleai/tests/common_utils.py

File: responsibleai/tests/causal/test_causal_general.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+def test_causal_classification_01():
```
Comment (gaugup) [inline responsibleai/tests/causal/test_causal_general.py:28]: This test passes anyways so why do we need this test to run in our gates?

**Review by gaugup — CHANGES_REQUESTED** [review]

File: responsibleai/tests/causal/test_causal_general.py (hunk 1 above)
Comment (riedgar-ms) [reply responsibleai/tests/causal/test_causal_general.py]: Actually there was a variant already there. Have switched to using it.

File: responsibleai/tests/causal/test_causal_general.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                     min_tree_leaf_samples=2,
+                     max_tree_depth=1,
+                     skip_cat_limit_checks=False,
+                     categories="auto",
+                     n_jobs=1,
+                     verbose=1,
+                     random_state=100,
+                     )
+
+    rai_i.compute()
```
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply responsibleai/tests/causal/test_causal_general.py:45]: Fair point, although I did think there was some value in showing the 'working' one. Have removed.

File: responsibleai/tests/causal/test_causal_general.py (hunk 2 above)
Comment (riedgar-ms) [reply responsibleai/tests/causal/test_causal_general.py]: Fair point, although I did think there was some value in showing the 'working' one. Have removed.

File: responsibleai/tests/causal/test_causal_general.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply responsibleai/tests/causal/test_causal_general.py:45]: Sorry, had put a reply in the wrong thread.

**Review by gaugup — APPROVED** [review]

File: responsibleai/tests/causal/test_causal_general.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply responsibleai/tests/causal/test_causal_general.py:45]: Wasn't `validate_causal()` - that tried to add a causal analysis.

**Pull request closed without merging**